### PR TITLE
Adds support for checking pg server version number on startup

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Current valid range: `=12.x`.

### CODE REVIEW

File: pkg/pgmodel/extension.go [hunk 1]
```diff
@@ -20,6 +20,36 @@ var (
...
+	}
+	if err := checkExtensionsVersion(conn); err != nil {
+		return fmt.Errorf("extensions-version: %w", err)
+	}
+	return nil
+}
+
+func checkPgVersion(conn *pgx.Conn) error {
+	var versionString string
+	if err := conn.QueryRow(context.Background(), "SHOW server_version_num;").Scan(&versionString); err != nil {
```
Comment (cevian) [inline pkg/pgmodel/extension.go:36]: This should be "SHOW server_version;"  which will return a X.Y number like "12.2" -> convert to semver via "%s.0"
https://www.postgresql.org/support/versioning/

File: pkg/version/version.go [hunk 2]
```diff
@@ -36,6 +36,9 @@ var (
 	Version    = "0.1.0-beta.4.dev.1"
 	CommitHash = ""
 
+	PgVersionNumRange       = ">=110000.0.0 <129999.0.0" // Corresponds to range within pg 11.00.00 to pg 12.99.99
```
Comment (cevian) [inline pkg/version/version.go:39]: we don't support 11  so this should be "=12.x" after the change I propose above

**Review by cevian — CHANGES_REQUESTED** [review]

**Comment by Harkishen-Singh** [comment]
@cevian done!

File: pkg/version/version.go [hunk 3]
```diff
@@ -36,6 +36,9 @@ var (
 	Version    = "0.1.0"
 	CommitHash = ""
 
+	PgVersionNumRange       = "=12.x" // Corresponds to range within pg 11.00.00 to pg 12.99.99
```
Comment (cevian) [inline pkg/version/version.go:39]:
```suggestion
	PgVersionNumRange       = "=12.x" // Corresponds to range within pg 12.0 to pg 12.99
```

File: pkg/pgmodel/extension.go [hunk 4]
```diff
@@ -20,6 +20,36 @@ var (
...
+	}
+	versionString = fmt.Sprintf("%s.0", versionString)
+	v, err := semver.Parse(versionString)
+	if err != nil {
+		return fmt.Errorf("could not parse postgresql version string: %w", err)
+	}
+	if !version.VerifyPgVersion(v) {
+		log.Error(
+			"msg", "incompatible postgresql version", "supported server version number within",
+			fmt.Sprintf("%s to %s", strings.Split(version.PgVersionNumRange, " ")[0], strings.Split(version.PgVersionNumRange, " ")[1]),
```
Comment (cevian) [inline pkg/pgmodel/extension.go:47]: There is no range here so nothing to split. Let's just do:
```
log.Error(
			"msg", "incompatible postgresql version", "supported server version range", version.PgVersionNumRange)
```

**Review by cevian — CHANGES_REQUESTED** [review]

File: pkg/pgmodel/extension.go [hunk 5]
```diff
@@ -20,6 +20,33 @@ var (
...
+	if err := conn.QueryRow(context.Background(), "SHOW server_version;").Scan(&versionString); err != nil {
+		return fmt.Errorf("error fetching postgresql version: %w", err)
+	}
+	versionString = fmt.Sprintf("%s.0", versionString)
+	v, err := semver.Parse(versionString)
+	if err != nil {
+		return fmt.Errorf("could not parse postgresql version string: %w", err)
+	}
+	if !version.VerifyPgVersion(v) {
+		log.Error("msg", "incompatible postgresql version", "supported server version range", version.PgVersionNumRange)
```
Comment (cevian) [inline pkg/pgmodel/extension.go:45]: This should be returned as an error (fmt.Errorf) not simply logged.

File: pkg/pgmodel/extension.go [hunk 6]
```diff
@@ -20,6 +20,33 @@ var (
 	ExtensionIsInstalled = false
 )
 
+// checkVersions is responsible for verifying the version compatibility of installed Postgresql database and extensions.
+func checkVersions(conn *pgx.Conn) error {
+	if err := checkPgVersion(conn); err != nil {
+		return fmt.Errorf("pg-version: %w", err)
```
Comment (cevian) [inline pkg/pgmodel/extension.go:26]:
```suggestion
		return fmt.Errorf("Problem checking PostgreSQL version: %w", err)
```

File: pkg/pgmodel/extension.go [hunk 7]
```diff
@@ -20,6 +20,33 @@ var (
...
+	}
+	if err := checkExtensionsVersion(conn); err != nil {
+		return fmt.Errorf("extensions-version: %w", err)
```
Comment (cevian) [inline pkg/pgmodel/extension.go:29]:
```suggestion
		return fmt.Errorf("Problem checking Promescale extension version: %w", err)
```

**Review by cevian — CHANGES_REQUESTED** [review]

**Review by cevian — APPROVED** [review]

File: pkg/pgmodel/extension.go [hunk 8]
```diff
@@ -209,3 +236,15 @@ func getNewExtensionVersion(availableVersions semver.Versions, validRange semver
 	}
 	return semver.Version{}, false
 }
+
+// formatPgVersion removes the leading zeros in the passed version number.
+func formatPgVersion(s string) string {
+	var tmp string
```
Comment (cevian) [inline pkg/pgmodel/extension.go:242]: I think this should use `strings.TrimLeft(s,"0")`

File: pkg/pgmodel/extension.go [hunk 9]
```diff
@@ -20,6 +20,33 @@ var (
...
+	}
+	return nil
+}
+
+func checkPgVersion(conn *pgx.Conn) error {
+	var versionString string
+	if err := conn.QueryRow(context.Background(), "SHOW server_version_num;").Scan(&versionString); err != nil {
+		return fmt.Errorf("error fetching postgresql version: %w", err)
+	}
+	versionString = fmt.Sprintf("%s.%s.0", versionString[:2], formatPgVersion(versionString[4:]))
```
Comment (cevian) [inline pkg/pgmodel/extension.go:39]: Please add a comment about what this conversion is doing

**Review by cevian — APPROVED** [review]
Couple of nits left

**Pull request merged**